src/siren/Siren.js

```diff
@@ -12,13 +13,13 @@ export default function Siren(resource, baseUrl = '') {
             return deepArray(object)
         else if (isObject(object))
             return entity(object)
-        throw Errors.typeArg('object', 'array|object')
+        
+        true.should.be.false
```

Comment:
OK. I'd recommend using `throw new Error('Description of problem')` or similar for clarity.